src/table/utils/handle-key-press.ts

```diff
@@ -146,6 +169,40 @@ export const handleHeadKeyDown = ({
   }
 };
 
+/**
+ * handles ArrowDown and ArrowUp events for the head cell menu (move focus)
+ */
+export const handleHeadCellMenuKeyDown = (event: React.KeyboardEvent<HTMLLIElement>) => {
+  const { key, target } = event;
+  const currentFocus = document.activeElement ?? (target as HTMLElement);
+  // The rest key are handled by handleKeyDown in MUIMenuList
+  if (key === 'ArrowDown') {
+    // Prevent scroll of the page
+    // Stop triggering handleKeyDown in MUIMenuList
+    preventDefaultBehavior(event);
+    let nextFocusItem = getNextFocusItem(currentFocus);
+    while (nextFocusItem) {
+      if (nextFocusItem.ariaDisabled === 'true') {
+        nextFocusItem = getNextFocusItem(nextFocusItem);
+      } else {
+        (nextFocusItem as HTMLElement).focus();
+        break;
+      }
+    }
+  } else if (key === 'ArrowUp') {
+    preventDefaultBehavior(event);
+    let previousFocusItem = getPreviousFocusItem(currentFocus);
+    while (previousFocusItem) {
+      if (previousFocusItem.ariaDisabled === 'true') {
+        previousFocusItem = getPreviousFocusItem(previousFocusItem);
+      } else {
+        (previousFocusItem as HTMLElement).focus();
+        break;
+      }
+    }
+  }
+};
```

Comment:
A lot of duplicate code here, I would do it like this
```suggestion
  if (key === 'ArrowDown' || key === 'ArrowUp') {
    const getNewItem = key === 'ArrowDown' ? () => getNextFocusItem(currentFocus) : () => getPreviousFocusItem(currentFocus)
    // Prevent scroll of the page
    // Stop triggering handleKeyDown in MUIMenuList
    preventDefaultBehavior(event);
    let newFocusItem = getNewItem(currentFocus);
    while (newFocusItem) {
      if (newFocusItem.ariaDisabled === 'true') {
        newFocusItem = getNewItem(nextFocusItem);
      } else {
        (newFocusItem as HTMLElement).focus();
        break;
      }
    }
  }
```

Reply:
Nice one